mh/session.py

```diff
@@ -309,16 +309,18 @@ def try_transfer_circle_leadership(self):
             return None, None
 
         circle = self.get_circle()
-        if circle.leader != self or circle.get_population() <= 1 or not circle.departed:
+        with circle.lock():
```

Comment:
Can't you lock both at the same time?
```python
with circle.lock() and circle.player.lock():
```

Reply:
Done, only using a comma rather than an `and`, as per https://docs.python.org/3/reference/compound_stmts.html#the-with-statement